kv/dist_sender.go

```diff
@@ -650,6 +656,7 @@ func (ds *DistSender) Send(_ context.Context, call proto.Call) {
 
 	curReply := finalReply
 	for {
+		ds.tracer.Trace(args.Header(), "ds_send")
```

Comment:
It's currently legal, but rare. A `Trace` without a `Begin` will immediately finalize (so you get a single-item trace chunk). So as long as a request passed at least one Begin/Finalize block, it'll be in that block, otherwise it's its own Trace (example: commands popping out of Raft). That seems like a good way of handling things to me.